src/profile-logic/marker-schema.js

```diff
@@ -564,3 +570,33 @@ export function formatMarkupFromMarkerSchema(
       throw new Error(`Unknown format type ${JSON.stringify((format: empty))}`);
   }
 }
+
+/**
+ * Takes a marker and a RegExp and checks if any of its `searchable` marker
+ * payload fields match the search regular expression.
+ */
+export function markerPayloadMatchesSearch(
+  markerSchema: MarkerSchema,
+  marker: Marker,
+  searchRegExp: RegExp
+): boolean {
```

Comment:
I wonder if we should set `searchRegExp.lastIndex = 0` here too. Currently we should be good, but I'm concerned about how we might use this function in the future too.

Reply:
Yeah, that makes sense. Added it.